cfme/intelligence/reports/menus.py

```diff
@@ -198,3 +247,12 @@ def step(self, *args, **kwargs):
             "All EVM Groups",
             self.obj.group
         )
+
+
+@navigator.register(ReportMenusCollection, "All")
+class ReportMenus(CFMENavigateStep):
+    VIEW = AllReportMenusView
+    prerequisite = NavigateToAttribute("appliance.server", "CloudIntelReports")
+
+    def step(self, *args, **kwargs):
+        self.view.edit_report_menus.tree.click_path("All EVM Groups")
```

Comment:
This is a bit of an odd pattern - using the local view to navigate from prerequisite - if I can use widgets on the view that should result from this step method, why is the step method necessary?

Is `edit_report_menus` available in `self.prerequisite_view` for use here? I see its not defined in `AllReportMenusView` directly.

Reply:
It is defined in the prerequisite_view itself.

> why is the step method necessary?
I am not quite sure how to explain this, but I think if you understand why the `AllReportMenusView` view was added, it might help. It has been defined to assert flash messages, this was not checked earlier. `edit_report_menus` is defined in `CloudIntelReportsView`(parent_view), so it can be accessed by both `view` and `prerequisite_view`. Step method is necessary to click the `edit_report_menus` tree.

Reply:
I was asking mostly rhetorically :)

A navigation step should use the prerequisite view to navigate, not the view that's the result of executing the `step` method.